src/diagram/components/diagram.tsx

```diff
@@ -85,26 +85,49 @@ export const _Diagram = ({ dqRoot, getDiagramExport, hideControls, hideNavigator
     }
   };
 
+  const deleteEdge = (edge: Edge) => {
+    const { source, target } = edge;
+    const targetModel = dqRoot.getNodeFromVariableId(target);
+    const sourceModel = dqRoot.getNodeFromVariableId(source);
+    sourceModel?.tryVariable && targetModel?.removeInput(sourceModel.variable);
+  };
+
+  const deleteAllNodesEdges = (variableId: string) => {
+    const nodesEdges = dqRoot.reactFlowElements.filter((e) => {
+      return isEdge(e) && (e.source === variableId || e.target === variableId);
+    });
+    for (const edge of nodesEdges as Edge[]) {
+      deleteEdge(edge);
+    }
+  };
+
+  // onElementsRemove is called when the user's keyboard Delete key is pressed
+  // to delete a selected card or a selected connecting arrow (aka "edge").
   const onElementsRemove = (elementsToRemove: Elements) => {
     for (const element of elementsToRemove) {
       if (isEdge((element as any))) {
         const edge = element as Edge;
-        const { source, target } = edge;
-        const targetModel = dqRoot.getNodeFromVariableId(target);
-        const sourceModel = dqRoot.getNodeFromVariableId(source);
-        // sourceModel gets removed first when a node is selected to be deleted, otherwise, just remove the connection
-        sourceModel?.tryVariable && targetModel?.removeInput(sourceModel.variable);
+        deleteEdge(edge);
       } else {
-        // If this is the selected node we need to remove it from the state too
         const nodeToRemove = dqRoot.getNodeFromVariableId(element.id);
-        if (dqRoot.selectedNode === nodeToRemove) {
-          dqRoot.setSelectedNode(undefined);
-        }
```

Comment:
I found this step didn't seem necessary. Even without it, once a card is deleted the `dqRoot.selectedNode` becomes `undefined`  automatically.